qutebrowser/misc/editor.py

```diff
@@ -121,21 +122,72 @@ def edit(self, text):
             return
 
         self._remove_file = True
-        self._start_editor()
+
+        # Here we calculate the line and column of the caret based on its
+        # position and the given text.
+        #
+        # NOTE: Both line and column are 1-based indexes, because that's what
+        # most editors use as line and column starting index.
+        # By "most" we mean at least vim, nvim, gvim, emacs, atom, sublimetext,
+        # notepad++, brackets, visual studio, QtCreator and so on.
+        #
+        # To find the line we just count how many newlines there are before
+        # the caret and add 1.
+        #
+        # To find the column we calculate the difference between the caret and
+        # the last newline before the caret.
+        #
+        # For example in the text `aaa\nbb|bbb` (| represents the caret):
+        # caret_position = 6
+        # text[:caret_position] = `aaa\nbb`
+        # text[:caret_psotion].count('\n') = 1
+        # caret_position - text[:caret_position].rfind('\n') = 3
+        #
+        # Thus line, column = 2, 3, and the caret is indeed in the second
+        # line, third column
+        line = text[:caret_position].count('\n') + 1
```

Comment:
Please add some tests for that calculation to `tests/unit/misc/test_editor.py` (maybe easier if you move the caret calculation to its own function).